src/merkle_tree/mod.rs

```diff
@@ -167,4 +278,55 @@ mod test {
             ));
         }
     }
+
+    #[test]
+    fn test_from_stwo_proof() {
```

Comment:
Could we add also failing tests to cover the cases when assertions are not valid ?